### PR TITLE
Update dotty to Scala 3.0.0-M1

### PR DESCRIPTION
**Pull request by lvitaly**

### CODE REVIEW

**Comment by lvitaly** [comment]
Hi @neko-kai,
I have tried to add Scala 3.0.0-M1 to izumi for adding it to zio next. This is the original issue https://github.com/zio/zio/issues/4397. But I got stuck with empty sbt-dotty plugin after running sbtgen. I find out that the root cause is an unsupported 3.x version pattern in isDotty function of sbtgen. I created PR for fixing it (https://github.com/7mind/sbtgen/pull/106).
Could you pls review my changes and help with the release of sbtgen for supporting Scala 3.0.0-M1. And I will update my PR to izumi next.

**Comment by neko-kai** [comment]
@lvitaly Thank you for your contributions! `sbtgen-0.0.65` with your changes should be on the way to maven-central right now

**Comment by neko-kai** [comment]
@lvitaly Is this ready for merge? Also could you please regenerate build for jvm-only - `./sbtgen.sc` - we prefer to keep that version checked-in (for IDE support)

**Comment by lvitaly** [comment]
@neko-kai Yep. It looks ready to merge. All tests passed. But I've made a few changes to the `scala_3` package, and it would be great if somebody double-checks it.

File: izumi-reflect/izumi-reflect/src/main/scala_3/izumi/reflect/dottyreflection/Inspector.scala [hunk 1]
```diff
@@ -122,30 +122,26 @@ abstract class Inspector(protected val shift: Int) extends InspectorBase {
   }
 
   private def prefixOf(symbol: Symbol): Option[AppliedReference] = {
-    if (symbol.maybeOwner.isNoSymbol) {
+    val mo = symbol.maybeOwner
+    if (!mo.exists || mo.isNoSymbol || mo.isPackageDef || mo.isValDef) {
```
Comment (neko-kai) [inline izumi-reflect/izumi-reflect/src/main/scala_3/izumi/reflect/dottyreflection/Inspector.scala]: @lvitaly `mo.isValDef` is a correct case I think, izumi-reflect does count value prefixes, e.g. for `trait A { trait B }; val a = new A{}; Tag[a.B]` the prefix is `a`

File: izumi-reflect/izumi-reflect/src/test/scala/izumi/reflect/test/CurrentDottySupportExtentTest.scala [hunk 2]
```diff
@@ -18,60 +18,52 @@ trait CurrentDottySupportExtentTest extends TagAssertions {
   trait Invariantoid[V]
   trait SubInvariantoid[V] extends Invariantoid[V] with Traitoid
 
-  // FIXME: report upstream: LTT macro calls do not work inside class body / inside normal test
```
Comment (neko-kai) [inline izumi-reflect/izumi-reflect/src/test/scala/izumi/reflect/test/CurrentDottySupportExtentTest.scala:21]: Odd that it works now, the principal fix is not merged yet (https://github.com/lampepfl/dotty/pull/9984), but that's good nonetheless.

**Review by neko-kai — APPROVED** [review]
One change left (I think `mo.isValDef` is a spurious filter as we're okay with value prefixes) and we can merge and cut a new version for 3.0.0-M1. Thanks! 👍

File: izumi-reflect/izumi-reflect/src/main/scala_3/izumi/reflect/dottyreflection/Inspector.scala (hunk 1 above)
Comment (lvitaly) [reply izumi-reflect/izumi-reflect/src/main/scala_3/izumi/reflect/dottyreflection/Inspector.scala]: @neko-kai I believe you are right. I checked `inspectSymbol` function and didn't find case for `ValDef`. That's why I excluded it by `mo.isValDef`. If I get you right then `case v: ValDef => NameReference(v.name)` would be enough. Or I'm wrong? Please advise me on how to proceed.

Comment (neko-kai) [reply izumi-reflect/izumi-reflect/src/main/scala_3/izumi/reflect/dottyreflection/Inspector.scala]: @lvitaly Yep, that would work correctly.

**Comment by lvitaly** [comment]
@neko-kai Done. I think this is ready for merge.

**Review by neko-kai — APPROVED** [review]

**Comment by neko-kai** [comment]
@lvitaly Released as [1.0.0-M10](https://github.com/zio/izumi-reflect/releases/tag/v1.0.0-M10)

**Pull request merged**